packages/compass-sidebar/src/modules/is-genuine-mongodb-visible.ts

```diff
@@ -12,15 +13,20 @@ export const TOGGLE_IS_GENUINE_MONGODB_VISIBLE =
   `${PREFIX}/TOGGLE_IS_GENUINE_MONGODB_VISIBLE` as const;
 interface ToggleIsGenuineMongoDBVisibleAction {
   type: typeof TOGGLE_IS_GENUINE_MONGODB_VISIBLE;
+  connectionId: ConnectionInfo['id'];
   isVisible: boolean;
 }
 export type IsGenuineMongoDBVisibleAction = ToggleIsGenuineMongoDBVisibleAction;
-export type IsGenuineMongoDBVisibleState = boolean;
+export type IsGenuineMongoDBVisibleState = Record<
+  ConnectionInfo['id'],
+  boolean
+>;
```

Comment:
this reminds me that we don't have this functionality in the new sidebar - do we have a ticket for it?

if I get this correctly, this state is for the modal, not the marker. I assume that with multiple connections there might be multiple markers, but only one modal at a time -> so this state doesn't need to be a record. it can be `IsGenuineMongoDBVisibleState = ConnectionInfo['id'] | undefined`.

see:
```
isGenuineMongoDBVisible:
      state.isGenuineMongoDBVisible[initialConnectionInfo.id], // this is for the modal
isGenuine:
      state.instance[initialConnectionInfo.id]?.genuineMongoDB.isGenuine, // this would be for the marker
```

Reply:
Yeah so I added a follow up ticket to add the non genuine marker. Right now it's a Record<string, boolean> because multiple connections could be using non genuine MongoDB servers, so we would want to mark both of them.

Reply:
for marking multiple connections we use `isGenuine`.  `isGenuineMongoDBVisible` is used for `NonGenuineWarningModal`, which is a single piece:
https://github.com/mongodb-js/compass/pull/5677/files#diff-c49bde1f435db7e728d2d67930f0cc4bce9ca7c919c25a6f89c0dd1d768c0deaL206
the variable name is confusing, it should probably be `isNonGenuineWarningModalOpen`

Reply:
Ah, I get it now. Yeah you are right, probably the isNonGenuineWarningModalOpen can be a single variable. I need to think about this, because even if the naming is kind of misleading (I can change it, it's fine), I personally like the idea of being it one variable per connection, in case later we don't show a modal but something else (like a marker in the connection name or something)

Reply:
we do! this modal opens when we click on one of the markers. but perhaps you're right, in the future maybe the modal will contain something specific to the connection, then we could use the id. right now it's just a disclaimer with a link to https://www.mongodb.com/docs/compass/master/faq/#why-am-i-seeing-a-warning-about-a-non-genuine-mongodb-server-